### ISSUE TITLE
Documentation needs updating to activate the virtual environment created by tox

### ISSUE THREAD

**Issue by thompson318** [UCL/scikit-surgerybard#27]
In GitLab by @MianAhmad on Jul 3, 2019, 10:45

Considering the variable familiarity of students with virtualenv, I think it may be simpler to activate the virtual environment created by `tox` instead of creating a virtual environment before hand.

This [page](https://scikit-surgerybard.readthedocs.io/en/latest/02_0_Build_Your_Own_Systems.html) is in referenced.

**Comment by thompson318** [comment]
In GitLab by @MianAhmad on Jul 3, 2019, 10:45

May be better to keep the following steps in the documentation  
Clone the repo.
```
git clone git@weisslab.cs.ucl.ac.uk:WEISS/SoftwareRepositories/SNAPPY/scikit-surgerybard.git
```
Change directory to the cloned repo
```
cd sciket-surgerybard
```
Run tox
```
tox
```
To activate the virtual environment on mac/linux created by tox
```
source .tox/py36/bin/activate
```
OR      
To activate the virtual environment on windows
```
cd .tox\py36\bin\
activate
```
It will give us `py36` virtual environment with all the required packages and everything working mostly. Also no need of privileges like `--user`.

**Comment by thompson318** [comment]
In GitLab by @MianAhmad on Jul 3, 2019, 10:49

changed the description

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Jul 3, 2019, 11:00

This would avoid any confusion with setting up virtual envs, but it might just add in different confusion, as I doubt any of the students would be familiar with what tox is/does.

We could add it as an optional step, in case any one has a problem getting the right packages installed.

**Comment by thompson318** [comment]
In GitLab by @StephenThompson on Jul 3, 2019, 11:36

I don't really understand why we're using tox and virtual environments for this. Is not pip install simpler?

**Comment by thompson318** [comment]
In GitLab by @MianAhmad on Jul 3, 2019, 13:53

Using virtualenv is to keep their system python intact. They can do the process again and again each time in new virtual environment starting from clean slate.   

Keeping the process independent from native python and packages is also important to keep all the apps on computer working. E.g. sciket-surgeryvtk doesn't work with the latest version of PySide2. If they have another application which needs latest version of PySide2 then they have to choose between one if not using virtualenv.

**Comment by thompson318** [comment]
In GitLab by @MianAhmad on Jul 3, 2019, 13:57

@ThomasDowrick yes we can keep it optional. I noticed that first of all they want the application to run without noticing tox or virtualenv or anything else. Anything working and in one go is what they like :)

**Comment by thompson318** [comment]
In GitLab by @MianAhmad on Jul 4, 2019, 11:03

mentioned in commit 2137640a882dd9cd09e6f8dc8982c0a95ecadcc5

**Comment by thompson318** [comment]
In GitLab by @MianAhmad on Jul 4, 2019, 11:05

mentioned in commit 389e22d2ff221bf169dedd129c0b66c4dbbf0bf6

**Comment by thompson318** [comment]
In GitLab by @MianAhmad on Jul 4, 2019, 11:21

mentioned in commit c499d63702c79b98cdb730056b6c5b3a9fb08ec0

**Comment by thompson318** [comment]
In GitLab by @MianAhmad on Jul 4, 2019, 11:43

mentioned in merge request !6

**Comment by thompson318** [comment]
In GitLab by @MattClarkson on Jul 4, 2019, 18:43

closed via commit c2439247c2dffe021fd8f0a54f3358d766a514fe
